pkg/rancher-desktop/main/update/MSIUpdater.ts

```diff
@@ -0,0 +1,154 @@
+import { spawn } from 'child_process';
+import fs from 'fs';
+import path from 'path';
+
+import { AllPublishOptions, newError } from 'builder-util-runtime';
+import { NsisUpdater } from 'electron-updater';
+import { InstallOptions } from 'electron-updater/out/BaseUpdater';
+import { ElectronHttpExecutor } from 'electron-updater/out/electronHttpExecutor';
+import { findFile } from 'electron-updater/out/providers/Provider';
+import { verifySignature } from 'electron-updater/out/windowsExecutableCodeSignatureVerifier';
+import { Lazy } from 'lazy-val';
+
+import mainEvents from '@/main/mainEvents';
+import paths from '@/utils/paths';
+
+import type { AppAdapter } from 'electron-updater/out/AppAdapter';
+import type { DownloadUpdateOptions } from 'electron-updater/out/AppUpdater';
+
+/**
+ * MsiUpdater implements updating for Rancher Desktop's MSI-based installer.
+ */
+// We extend from NsisUpdater because extending BaseUpdater appears to cause
+// issues with AppImageUpdater (where it thinks BaseUpdater is undefined)?
+export default class MsiUpdater extends NsisUpdater {
+  // eslint-disable-next-line no-useless-constructor -- This is used to change visibility
+  constructor(options?: AllPublishOptions | null, app?: AppAdapter) {
+    super(options, app);
+  }
+
+  // This implements an abstract method in BaseUpdater.
+  protected doDownloadUpdate(downloadUpdateOptions: DownloadUpdateOptions): Promise<string[]> {
+    const { info, provider } = downloadUpdateOptions.updateInfoAndProvider;
+    const fileInfo = findFile(provider.resolveFiles(info), 'msi');
+
+    if (!fileInfo) {
+      throw newError(`Could not find update information for MSI installer`,
```

Comment:
Would it be helpful to include `info` in the error? is there anything useful in the `info`?

Reply:
Yes, `info.version` would be useful. Added in b8397db37c3b8e845e0779fa7796e495c00df7fc.